docs/vgclog_examples.md

```diff
@@ -706,3 +707,895 @@ Analyzing the structure and elements of this example log output shows that this
+
+The global cycle is split into multiple increments, each recorded as `type="global mark phase"`. A global mark phase increment involves an STW subincrement, which runs a global mark operation during an STW pause, followed by a *global mark phase(GMP) work packet* subincrement. The`GMP work packet` subincrement involves a processing operation that runs concurrently. The `GMP work packet` subincrement might also use an STW pause to complete if the subincrement is interrupted by a partial or global cycle trigger.
+
+Splitting the global mark phase into these increments and subincrements reduces pause times by running the majority of the GC work concurrently and interleaving global mark phase increments with partial GC cycles, and, rarely, [global GC cycles](vgclog_examples.md#balanced-global-gc-cycle).
+
+A global *mark* GC cycle increment is triggered by an [allocation taxation threshold](vgclog_examples.md#balanced-examples).
+
+The following elements log the GC increments, subincrements and operations of the global *mark* GC cycle:
+
+|GC increment         | GC operations| *stop-the-world* or concurrent| XML element of GC increment| Details                         |
+|---------------------|-------------|-------------------------------|--------------------------------------|-----------------------|
+|`global mark phase` subincrement| mark | *stop-the-world* | `<gc-start>`, `<gc-end>` |The global mark phase operations start at the beginning of the cycle and run through all *regions* until the final *region* |
+|`GMP work packet processing` subincrement| work packet processing (WPP) operations | concurrent and sometimes final operations during a *STW* to complete the subincrement | `<concurrent-start>`, `<concurrent-end>`| The `GMP work packet processing subincrement` runs immediately after the `global mark phase` subincrement |
+|final global mark phase increment | final global mark phase operations including class unload | *stop-the-world* | `gc-start>`, `<gc-end>`| Final increment. Runs the final global mark phase operations followed by operations to finish the cycle  |
+
+The `balanced` global *mark* GC cycle follows a general structure in the verbose GC log as shown. The lines are indented to help illustrate the flow and some child elements are omitted for clarity:
+
+```xml
+<exclusive-start/>                        (STW pause starts)
+
+<allocation-taxation/>                    (memory threshold trigger recorded)
+
+<cycle-start type="global mark phase"/>   (global mark cycle starts)
+
+<gc-start type="global mark phase"/>      (1st GMP STW subincrement starts)
+
+    <mem-info>                            (memory status before operations)
+
+    <remembered-set>
+
+    </mem-info>         
+
+</gc-start> 
+
+<gc-op type="mark increment" />           (STW copy forward operation completed)
+
+<gc-op  type="class unload" />            (optional STW class unload operation completed)
```

Comment:
class unloading gc-op can really occur in the very last STW increment of GMP cycle

Reply:
Just to confirm - is this correct: the class unloading is optional in GMP increments, but an essential part of the final GMP increment?

Reply:
Class unloading is NOT optional for (==sweep, but sweep deals with objects, class unloading deals with objects). Only happens once global mark cycle is complete. Class unloading is NOT optioanl for non-final GMP increments. 

sweep can only happen if we have COMPLETE liveness operation.